core/vm/evm.go

```diff
@@ -63,12 +63,9 @@ func (evm *EVM) precompile(addr common.Address) (PrecompiledContract, bool) {
 		precompiles = PrecompiledContractsHomestead
 	}
 	p, ok := precompiles[addr]
-	// Restrict overrides to known precompiles
-	if ok && evm.chainConfig.IsOptimism() && evm.Config.OptimismPrecompileOverrides != nil {
-		override, ok := evm.Config.OptimismPrecompileOverrides(evm.chainRules, p, addr)
-		if ok {
-			return override, ok
-		}
+	if evm.Config.PrecompileOverrides != nil {
+		override := evm.Config.PrecompileOverrides(evm.chainRules, p, addr)
+		return override, override != nil
```

Comment:
I'm slightly nervous about exposing the complexity of fork handling to the overrides. With this API, it's really easy to mean to override a precompile but actually wind up adding it to a fork that didn't previously have it (see comment on monorepo PR). I can see how the extra flexibility can be useful though.

Reply:
Good catch, currently we only really have one critical consumer of this, adding protection there I think is sufficient. As you said, once we have more usages of it we can change the API to be more defensive against it by separating additions/replacements/deletions of precompiles.